### ISSUE TITLE
Change Elastic Agent behaviour when getting 404 on checkin calls

### ISSUE THREAD

**Issue by jlind23** [elastic/elastic-agent#2414]
We observed that when an agent is getting an `http 404` while performing a checkin it will retry forever.

We should change this behavior by putting in place a mechanism that will every 15min and not every second as of today.

**Comment by cmacknz** [comment]
> The best path forward here will be to shutdown the agent after X attempt in order to avoid infinite retry.

I don't think disabling the agent makes sense here. I'm actually not sure that shutting down on a 403 makes sense. The agent should never shut itself down unless explicitly instructed to. This feels like it goes against the tamper protection work we have going on.

Why is retrying forever on a 404 undesirable? We could make the backoff more aggressive to retry less frequently, like every 15 minutes.

**Comment by jlind23** [comment]
As of today Agent is retrying forever and call the api every second which at scale generate a lot of traffic on our end if we host fleet server. If we do not shut down the agent, this can continue forever even with an aggressive backoff in place.

**Comment by cmacknz** [comment]
Why are we getting 404s at all? Is that an expected result? What put the system into this state?

**Comment by jlind23** [comment]
If the agent document is removed due to either a manual deletion or a rollback to a previous ES snapshost, then we end up getting a 404.

**Comment by cmacknz** [comment]
Right, my preferred outcome would be to make the backoff on a 404 more aggressive with a longer maximum duration. Retrying every 15 minutes seems reasonable, definitely not every second which is what we do today. 

The agent has no way to know if the 404 is an intentional error, I don't think shutting down or unenrolling is a good way to handle this error.

As far as I can tell we don't shut down or do any special handling for 403s. We used to unenroll automatically when getting an invalid API key response but this was removed (likely because of predictably bad results when the API key error was a result of a bug). Here is this code in v8.3.3 for reference:

https://github.com/elastic/elastic-agent/blob/0ffbedfd669a998fe0cc5c4d0305338515f07c46/internal/pkg/agent/application/gateway/fleet/fleet_gateway.go#L337-L349
